### PR TITLE
Feature/85456044/build diligence branch

### PR DESCRIPTION
**Pull request by timz1290**

### CODE REVIEW

File: lib/app/app.rb [hunk 1]
```diff
@@ -107,6 +107,11 @@ def remote_ip
 
       STDERR.puts "BUILD: #{build.inspect}"
 
+      diligence_branch = Integrity::DiligenceBranch.first(:branch => build.branch)
```
Comment (rahmal) [inline lib/app/app.rb]: One of my favorite short cuts, is to do the ```if``` and the variable setting in one call. Especially if the variable is only going to be used in the if block:

```Ruby
if diligence_branch = Integrity::DiligenceBranch.first(:branch => build.branch)
  diligence_branch.rebuild_environment
end

File: lib/app/app.rb [hunk 2]
```diff
@@ -181,6 +186,45 @@ def remote_ip
...
 
+    get '/diligence_branches' do
+      load_projects
+      @active_diligence_branches = Integrity::DiligenceBranch.all(order: :id.desc, active: true)
+      @inactive_diligence_branches = Integrity::DiligenceBranch.all(order: :id.desc, active: false)
+      @branches = Integrity::Branch.all(limit:30, order: :last_build_id.desc)
+      show :diligence_branches
+    end
+
+    get '/diligence_branches/:branch_id/start' do |branch_id|
```
Comment (rahmal) [inline lib/app/app.rb]: Can you please change the ```build_id``` to ```id```, the way you have in the proceeding actions (or vice-versa) so they are consistent.

**Comment by rahmal** [comment]
CR'd Good work. Very clean.

**Comment by kenberland** [comment]
A key missing feature is the ability to delete a diligence branch.  Sure, you can make it inactive, but how do you delete it?

**Pull request merged**